pkg/image/api/helper.go

```diff
@@ -523,6 +525,8 @@ func ImageWithMetadata(image *Image) error {
 			return err
 		}
 
+		image.DockerImageManifestMediaType = schema2.MediaTypeManifest
```

Comment:
> should we only override this if it is empty?

I don't think so. This function is called after operations such as replacement of manifest. All the metadata needs to be updated based on that. This field depends on the manifest as well.